docs/source/API/containers/Array.rst

```diff
@@ -0,0 +1,173 @@
+* Unlike a C-style array, it doesn't decay to ``T*`` automatically.
+* As an aggregate type, it can be initialized with aggregate-initialization given at most ``N`` initializers that are convertible to ``T``: ``Kokkos::Array<int, 3> a = { 1, 2, 3 };``.
+
+..
+  The API of the entity.
+
+Interface
+---------
+
+..
+  The declaration or signature of the entity.
+
+.. cppkokkos:struct:: template <class T, size_t N> Array
+
+  ..
+    Template parameters (if applicable)
+    Omit template parameters that are just used for specialization/are deduced/ and/or should not be exposed to the user.
+
+  .. rubric:: Template Parameters
+
+  :tparam T: The type of the element being stored.
+  :tparam N: The number of elements being stored.
+
+  .. rubric:: Public Types
+
+  .. cppkokkos:type:: value_type = T
+  .. cppkokkos:type:: pointer = T*
+  .. cppkokkos:type:: const_pointer = const T*
+  .. cppkokkos:type:: reference = T&
+  .. cppkokkos:type:: const_reference = const T&
+  .. cppkokkos:type:: size_type = size_t
+  .. cppkokkos:type:: difference_type = ptrdiff_t
+
+  .. rubric:: Public Member Functions
+
+  .. cppkokkos:function:: static constexpr bool empty()
+
+    :return: ``N == 0``
+
+  .. cppkokkos:function:: static constexpr size_type size()
+  .. cppkokkos:function:: constexpr size_type max_size() const
+
+    :return: ``N``
+
+  .. cppkokkos:function:: template<class iType> constexpr reference operator[](const iType& i)
+  .. cppkokkos:function:: template<class iType> constexpr const_reference operator[](const iType& i) const
+
+    :tparam iType: An integral type or an unscoped enum type.
+
+    :return: A reference to the ``i``-th element of the array.
+
+  .. cppkokkos:function:: constexpr pointer data()
+  .. cppkokkos:function:: constexpr const_pointer data() const
+
+    :return: A pointer to the first element of the array.  If ``N == 0``, the return value is unspecified and not dereferenceable.
+
+
+Deduction Guides
+----------------
+
+.. cppkokkos:function:: template<class T, class... U> Array(T, U...) -> Array<T, 1 + sizeof...(U)>
+
+Non-Member Functions
+--------------------
+
+..
+  These should only be listed here if they are closely related. E.g. friend operators. However,
+  something like view_alloc shouldn't be here for view
+
+.. cppkokkos:function:: template<class T, size_t N> constexpr bool operator==(const Array<T, N>& l, const Array<T, N>& r) noexcept
+
+   :return: ``true`` if and only if ∀ the elements in ``l`` and ``r`` compare equal.
+
+.. cppkokkos:function:: template<class T, size_t N> constexpr bool operator!=(const Array<T, N>& l, const Array<T, N>& r) noexcept
+
+   :return: ``!(l == r)``
+
+.. cppkokkos:function:: template<class T, size_t N> constexpr kokkos_swap(Array<T, N>& l, Array<T, N>& r) noexcept(N == 0 || is_nothrow_swappable_V<T>)
+
+   :return: If ``T`` is swappable or ``N == 0``, each of the elements in `l` and `r` are swapped via ``kokkos_swap``.
```

Comment:
not really a return ... but might be to render it neatly in sphinx

Reply:
Yea.  I'm trying to both keep the C++ noise (SFINAE and a separate function definition for `kokkos_swap(Array<T, 0>&, Array<T, 0>&)) out of the function call documentation here, as well as describe the effects of the call.  Unfortunately, I couldn't find a Sphynx keyword other than :return: that lets us render this decently.